src/Listener/ApiProblemListener.php

```diff
@@ -63,13 +64,13 @@ public function attach(EventManagerInterface $events, $priority = 1)
 
     /**
      * @param EventManagerInterface $events
+     * @return void
      */
     public function detach(EventManagerInterface $events)
     {
         foreach ($this->listeners as $index => $listener) {
-            if ($events->detach($listener)) {
-                unset($this->listeners[$index]);
-            }
+            $events->detach($listener);
```

Comment:
`detach` doesn't return anything, so this check was actually preventing the listeners from being unset.